pyomo/contrib/pynumero/sparse/block_matrix.py

```diff
@@ -393,9 +392,8 @@ def tocoo(self, copy=True):
 
         # create pointers for COO matrix (row, col, data)
         data = np.empty(nonzeros, dtype=dtype)
-        idx_dtype = get_index_dtype(maxval=max(shape))
-        row = -np.ones(nonzeros, dtype=idx_dtype)
-        col = -np.ones(nonzeros, dtype=idx_dtype)
+        row = -np.ones(nonzeros, dtype=int)
+        col = -np.ones(nonzeros, dtype=int)
```

Comment:
Do we know when NumPy moved to just using `int` as the index dtype? Do we need to update the minimum numpy version?  Doesn't that place an upper limit of ~4B nonzeros?  (Is that a problem?)

Reply:
I'll have to check on the numpy minimum version. 

A dtype of `int` is really `int64`:
```
In [1]: import numpy as np

In [2]: a = np.zeros(3, dtype=int)

In [3]: a.dtype
Out[3]: dtype('int64')

In [4]: a[0] = 9.1e18

In [5]: a
Out[5]: array([9100000000000000000,                   0,                   0])
```

Reply:
Also, these are the row and column indices, so it is possible to have matrices of dimensions 9e18 by 9e18...

Reply:
Looks like this works all the way back to numpy 1.13 which is the current minimum version.

Reply:
Great!  Thanks!